src/codecs/webp/extended.rs

```diff
@@ -324,10 +372,13 @@ impl ExtendedImage {
     pub(crate) fn fill_buf(&self, buf: &mut [u8]) {
         match &self.image {
             // will always have at least one frame
-            ExtendedImageData::Animation { frames, .. } => &frames[0].image,
-            ExtendedImageData::Static(image) => image,
+            ExtendedImageData::Animation { .. } => {
+                if let Some(Ok(frame)) = self.as_frames().into_iter().nth(0) {
```

Comment:
This should probably assert that frames isn't empty, as the old code does

Reply:
I don't understand, you want it to crash if frames is empty?

Reply:
The current code does `frames[0]`, so it crashes if frames is empty. Which based on the code comment should never happen. It is better to fail loudly if something unexpected happens rather than just silently doing nothing

Reply:
That is entirely debatable. I default to code that should not crash. (Kind of the idea of Rust me thinks)

I actually don't care at all how it's fixed but this code currently crashes a lot of programs due to the buffer size mismatch and it needs fixed.

Reply:
The idea is to _handle_ errors. Anyway, I'd like to avoid taking in a PR that fixes one bug but accidentally conceals another one. So for now, could you add an `assert!` or `expect` or similar? That and the cargo test failure, and I think this PR is ready to merge